### DISCUSSION TITLE
Having issue adding screenshot to the allure report,

### DISCUSSION THREAD

**Discussion by anandseshadri** [webdriverio/webdriverio#8333] [Q&A]
Hi, I was trying to add a screenshot to the allure report that is generated while testing electron, my configuration is as follows 

`"@wdio/allure-reporter": "^7.19.7"`
**wdio.confi.js**

```
framework: 'mocha',
  mochaOpts: {
    ui: 'bdd',
    timeout: 3000,
  },
reporters: [['allure', 
  {
    outputDir: 'allure-results',
    disableWebdriverStepsReporting: true,
    disableWebdriverScreenshotsReporting: false
}]],
//this not getting executed
afterStep: function (step, scenario, { error, duration, passed }, context) {
  if (error) {
    browser.takeScreenshot();
  }
}
```
I have tried using the `afterCommand` and  `afterTest`, but The `browser.takeScreenshot()` is not capturing the Screenshot and I don't see the error. Why the `browser.takeScreenshot()` is not working

**Comment by portyu9** [comment]
Hi @anandseshadri 

The main issue is the hook: your config uses **Mocha**, but `afterStep` is a **Cucumber-only** WebdriverIO hook. It will not run for Mocha tests.

Use `afterTest` instead, and make the screenshot call asynchronous:

```js
afterTest: async function (test, context, { error }) {
    if (error) {
        await browser.takeScreenshot()
    }
}
```

You already have:

```js
disableWebdriverScreenshotsReporting: false
```

so the Allure reporter can attach screenshots produced through the WebDriver screenshot command.

If you prefer to make the attachment explicit:

```js
import allureReporter from '@wdio/allure-reporter'

afterTest: async function (test, context, { error }) {
    if (!error) {
        return
    }

    const screenshot = await browser.takeScreenshot()

    allureReporter.addAttachment(
        'Failure screenshot',
        Buffer.from(screenshot, 'base64'),
        'image/png'
    )
}
```

So `browser.takeScreenshot()` itself isn't the primary problem shown here — the code was placed in a hook that doesn't run with the configured framework, and the asynchronous command wasn't being awaited.

I reviewed the entire configuration in the post. There were no replies, screenshots, or additional attachments.

Please mark this answer as accepted if it helped, thank you.
